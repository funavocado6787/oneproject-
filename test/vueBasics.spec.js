import { RouterLinkStub, shallowMount, createLocalVue, mount } from '@vue/test-utils'
import VueBasics from '../pages/VueBasics'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Vuetify)
import DisplayCode from '../components/displayCode.vue'
Vue.component(DisplayCode.name, DisplayCode);

const localVue = createLocalVue()
localVue.use(Vuex)

describe('vueBasics.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            vueBasicsSnippets: {}
        }
        store = new Vuex.Store({
            modules: {
                codeSnippet: {
                    namespaced: true,
                    actions: {
                        loadVueBasicsSnippets({commit}) {
                            commit('setVueBasicsSnippets', {
                                basicComponent: 'basicComponent',
                             });
                        }
                    },
                    getters: {
                        getVueBasicsSnippets: state => state.vueBasicsSnippets
                    },
                    state,
                    mutations: {
                        setVueBasicsSnippets(state, data) {
                            state.vueBasicsSnippets = data;
                        }
                    }
                }
            }
        })
    })

    test('Smoke Test - check page renders by looking for some hard coded text', async () => {
        const wrapper = shallowMount(VueBasics, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Components');
    })

    test('vueBasics snippet load call was made on creation', async () => {
        const wrapper = shallowMount(VueBasics, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Components');
        expect(Object.keys(state.vueBasicsSnippets).length).toBe(1);
    })

    test('vueBasics made use of basicComponent', async () => {
        const wrapper = mount(VueBasics, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('basicComponent');
    })

    test('match snapshot', async () => {
        const wrapper = mount(VueBasics, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})













