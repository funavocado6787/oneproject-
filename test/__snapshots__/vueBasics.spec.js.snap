// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vueBasics.vue match snapshot 1`] = `
<div
  class="container layout column"
>
  <h3>
    Components
  </h3>
   
  <p>
    The basic component framework
  </p>
   
  <div
    class="container d-flex flex-row justify-start"
  >
    <div
      class="container"
    >
      <div
        id="app"
      >
        <div
          class="prism-editor-wrapper my-editor"
        >
          <div
            class="prism-editor__container"
          >
            <textarea
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="prism-editor__textarea"
              data-gramm="false"
              data-testid="textarea"
              placeholder=""
              spellcheck="false"
            />
            <pre
              class="prism-editor__editor"
              data-testid="preview"
            >
              basicComponent
              <br />
            </pre>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="container"
    >
      <h4>
        Imports
      </h4>
       
      <p>
        
        Other files that this component needs. If it is a component that is
        common use throughout the application it is better to put it in
        main.js. If using Nuxt the reference is made automatically by put it
        in the component folder.
      
      </p>
       
      <h4>
        Template
      </h4>
       
      <p>
        
        The template is the html file that will be rendered. This can be put
        in the component or a serperate file.
      
      </p>
       
      <h4>
        Props
      </h4>
       
      <p>
        
        The calling component can push data into the child component via the
        html attributes. In the child component this is accessible through
        props in array or object syntax. Object syntax is the Vue Style guides
        suggestion.
      
      </p>
       
      <h4>
        Methods
      </h4>
       
      <p>
        
        This is the place that the component has its custom methods. This
        could be for loading data, user interaction.
      
      </p>
       
      <h4>
        Computed
      </h4>
       
      <p>
        
        Intended for simple operations and reducing concatenations and other
        manipluation within the page. A useful one for custom validation
        methods when using vee-validate.
      
      </p>
       
      <h4>
        Watch
      </h4>
       
      <p>
        
        If you need interation when a value or object changes this is the
        place for it.
      
      </p>
       
      <h4>
        Created
      </h4>
       
      <p>
        
        This is part of the page life cycle it runs on the component before
        the DOM has loaded. This is the best place to load initial data.
        
        <br />
        While Created and Mounted are the most commonly used lifecycle
        hooks there is also a beforeCreate, beforeMount, beforeUpdate,
        updated, beforeDestroy and destroyed. More information on these is
        available
        
        <a
          href="https://alligator.io/vuejs/component-lifecycle/"
        >
           here 
        </a>
        .
      
      </p>
       
      <h4>
        Mounted
      </h4>
       
      <p>
        
        This is part of the page life cyle where the DOM is added. This should
        not be used for fetching data.
      
      </p>
    </div>
  </div>
</div>
`;
